### PR TITLE
Security checking of WASI-NN

### PR DESCRIPTION
**Pull request by Ahmedounet**
Added security checks for memory boundary conditions on wasi-nn functions

### CODE REVIEW

File: core/iwasm/libraries/wasi-nn/wasi_nn.c [hunk 1]
```diff
@@ -47,81 +59,63 @@ wasi_nn_init_execution_context()
...
     { #func_name, wasi_nn_##func_name, signature, NULL }
 /* clang-format on */
 
 static NativeSymbol native_symbols_wasi_nn[] = {
-    REG_NATIVE_FUNC(load, "(ii)i"),
-    REG_NATIVE_FUNC(set_input, "(ii*i*)i"),
-    REG_NATIVE_FUNC(get_output, "(ii*ii*i)i" ),
-    REG_NATIVE_FUNC(compute, "(i)i")
+    REG_NATIVE_FUNC(load, "(ii)i"), REG_NATIVE_FUNC(set_input, "(iiiii)i"),
+    REG_NATIVE_FUNC(get_output, "(ii*~)i"), REG_NATIVE_FUNC(compute, "(i)i")
```
Comment (tonibofarull) [inline core/iwasm/libraries/wasi-nn/wasi_nn.c:118]: why `set_input` is removed?

Comment (Ahmedounet) [reply core/iwasm/libraries/wasi-nn/wasi_nn.c:118]: not removed, it's there, just the clang format conventions !

Comment (tonibofarull) [reply core/iwasm/libraries/wasi-nn/wasi_nn.c:118]: oh true!

**Review by tonibofarull — APPROVED** [review]

**Review by tonibofarull — CHANGES_REQUESTED** [review]
`wasi_nn_init_execution_context` should call `_init_execution_context` inside the tensorflow specification which will call:
```
interpreter->AllocateTensors();
```

**Pull request merged**